### DISCUSSION TITLE
是否能把行头列头设置成可自选的模式？是否能将合并单元格的功能做成真正的将合并后的单元格的数值进行相加，而不是只是简单的显示 附图

### DISCUSSION THREAD

**Discussion by annisFromchina** [antvis/S2#2797] [Ideas]
是否能把行头列头设置成可自选的模式？我自己P了一个简易图
![ideaOfAnTvS201](https://github.com/antvis/S2/assets/174005360/af4e6162-c4ef-46f1-bf5f-0c1b9c4090ab)

是否能将合并单元格的功能做成真正的将合并后的单元格的数值进行相加，而不是只是简单的显示
![ideaOfAnTvS202](https://github.com/antvis/S2/assets/174005360/3d0d3bac-a98b-4db8-b83a-0d366ab4e0b1)

**Comment by lijinke666** [comment]
> 是否能把行头列头设置成可自选的模式

目前没开放配置, 感兴趣可以来个 PR, 比较简单.

https://github.com/antvis/S2/blob/73b9b6cd7f49fcf4d11e641e76059adfbf77e4e1/packages/s2-react/src/components/switcher/dimension/index.tsx#L73-L74

> 是否能将合并单元格的功能做成真正的将合并后的单元格的数值进行相加，而不是只是简单的显示

合并单元格的能力比较弱, 人力有限, 不是后续迭代的重点, 目前是是参考的 Excel 的行为 保留左上角的文字, 如你需要求和, 可以参考 [自定义合并单元格](https://s2.antv.antgroup.com/examples/custom/custom-cell/#custom-merged-cell) 的示例.

```tsx
class CustomMergedCell extends MergedCell {
  // 重写绘制文本
  getFieldValue() {
    const [groupA] = this.spreadsheet.options.mergedCellsInfo || []
    
    // 根据行列索引拿到合并之前单元格的值
    const fieldValues = groupA.map(({rowIndex, colIndex}) => {
      const meta = this.spreadsheet.facet.getCellMeta(rowIndex, colIndex)
      return meta.fieldValue
    })

    const sum = fieldValues.reduce((a, b) => a + b, 0)

    return sum
  }

  getTextPosition() {
    const defaultTextPosition = super.getTextPosition()
    const { x, y, width, height } = this.meta

    console.log(this.meta)

    return {
      ...defaultTextPosition,
      // 重写文本位置
      // x: 0,
      // y: 0
    }
  }
}
```

![image](https://github.com/antvis/S2/assets/21015895/f4e30979-27e3-4743-9dd5-01023daa460f)

![image](https://github.com/antvis/S2/assets/21015895/3bfc6db0-f36c-44ea-ad7e-a103a9dbec4c)



如果后续反馈的人较多, 会考虑内置, 欢迎 PR.
